app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -450,7 +450,7 @@ QueryBuilder makeQuery(String prefix, String studyId, AccountId accountId, Accou
                         "number", unguarded.getPhone().getNumber(),
                         "regionCode", unguarded.getPhone().getRegionCode());
             } else {
-                builder.append("AND acct.externalId=:externalId", "externalId", unguarded.getExternalId());
+                builder.append("AND (acctSubstudy.externalId=:externalId OR acct.externalId=:externalId)", "externalId", unguarded.getExternalId());
```

Comment:
We should update integ tests to verify that this new query works.

Reply:
Checked in some new PRs to verify this. Created an account with two external IDs and verified both can be used to retrieve the account. Only possible if it looks at the associative records.